osgp/platform/osgp-adapter-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/application/mapping/GridMeasurementPublishedEventConverter.java

```diff
@@ -36,30 +45,84 @@
 
     private static final Logger LOGGER = LoggerFactory.getLogger(GridMeasurementPublishedEventConverter.class);
 
-    private static final int SIMPLE_END_INDEX = 10;
+    private ObjectMapper objectMapper = new ObjectMapper();
 
     @Override
     public GridMeasurementPublishedEvent convert(final String source,
             final Type<? extends GridMeasurementPublishedEvent> destinationType, final MappingContext mappingContext) {
 
-        StringArrayToAnalogList stringArrayToAnalogList = null;
-        final String[] values = source.split(";");
-        if (values.length == SIMPLE_END_INDEX) {
-            stringArrayToAnalogList = new SimpleStringToAnalogList();
-        } else if (values.length == LsPeakShavingMeasurementType.getNumberOfElements() + 1) {
-            stringArrayToAnalogList = new LsMeasurementMessageToAnalogList();
-        } else {
-            LOGGER.error("String '{}' does not have the expected amount of fields, abandoning conversion", source);
+        LsMeasurementMessageToAnalogList stringArrayToAnalogList = null;
+        LOGGER.info("Source string: {}", source);
+
+        try {
+            final Payload payload = this.objectMapper.readValue(source, Payload[].class)[0];
```

Comment:
In case there is no 0'th element, a nullpointer exception will be thrown. Add a check for this.